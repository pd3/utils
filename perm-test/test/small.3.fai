1	20000
2	20000
3	20000
