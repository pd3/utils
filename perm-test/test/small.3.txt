

./perm-test -t test/small.3.tgt.txt -b test/small.3.bg.txt -f test/small.3.fai -c test/small.3.calls.txt -n 1e9

    fai:
        1	20000
        2	20000
        3	20000
    bg:
        1	1	10000
        2	1	10000
        3	1	10000
    tgt:
        1	1000	2000
        1	5000	6000
        2	1000	2000
        2	5000	6000
        3	1000	2000
        3	5000	6000

    BG	1	1	    999
    TGT	1	1000	2000
    BG	1	2001	4999
    TGT	1	5000	6000
    BG	1	6001	10000
    BG	2	1	    999
    TGT	2	1000	2000
    BG	2	2001	4999
    TGT	2	5000	6000
    BG	2	6001	10000
    BG	3	1	    999
    TGT	3	1000	2000
    BG	3	2001	4999
    TGT	3	5000	6000
    BG	3	6001	10000

    calls:
        1	1500	1500

    exact: 3.*(2000-1000+1 + 6000-5000+1)/(3*10000) = 0.2002
    perm-test:  0.200182947
    err: -1.7053e-05


./perm-test -t test/small.3.tgt.txt -b test/small.3.bg.txt -f test/small.3.fai -c test/small.3.calls.txt -n 1e9 --no-bg-overlap

