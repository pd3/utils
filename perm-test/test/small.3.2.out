# NITER_ROUNDS:
#    - number of iterations total
#    - number of batches
# TEST_ENR:
#    - number of iterations
#    - number of times the simulation had the same or more hits than observed in input data
#    - P-value derived from the two fields
# TEST_DPL:
#    - number of iterations
#    - number of times the simulation had the same or fewer hits than observed in input data
#    - P-value derived from the two fields
# TEST_FOLD:
#    - number of hits in the input data
#    - average number of hits in simulations
#    - average of standard deviations approximated from each batch using the current average estimate
SEED	1533323562
NITER_ROUNDS	1.000000e+05	1
TEST_ENR	100000	20202	2.020200e-01
TEST_DPL	100000	100000	1.000000e+00
TEST_FOLD	1	0.202020	0.401507
